msal/src/internal/java/com/microsoft/identity/client/MsalUtils.java

```diff
@@ -277,7 +279,7 @@ static String getChromePackage(final Context context) {
      * @return The Map of the items decoded with the given delimiter.
      */
     static Map<String, String> decodeUrlToMap(final String url, final String delimiter) {
-        final Map<String, String> decodedUrlMap = new HashMap<>();
+        final Map<String, String> decodedUrlMap = new LinkedHashMap<>();
```

Comment:
If the order is to be preserved, should we declare a more explicit return type for the method so that callers know? Otherwise, this behavior might be relied upon and if we refactor this implementation later we might unexpectedly break something

`static LinkedHashMap<String, String> decodeUrlToMap(...)`